File: Application/src/main/java/com/shareyourproxy/api/service/SharedLinkService.java
package com.shareyourproxy.api.service;

import com.shareyourproxy.api.domain.model.SharedLink;

import java.util.HashMap;

import retrofit.http.Body;
import retrofit.http.DELETE;
import retrofit.http.GET;
import retrofit.http.PUT;
import retrofit.http.Path;
import rx.Observable;

/**
 * Add and remove {@link SharedLink}s.
 */
public interface SharedLinkService {

    @GET("/shared.json")
    Observable<HashMap<String, SharedLink>> getSharedLinks();

    /**
     * add a {@link SharedLink}
     *
     * @param sharedId shared link identifier
     */
    @PUT("/shared/{sharedId}.json")
    Observable<SharedLink> addSharedLink(@Path("sharedId") String sharedId, @Body SharedLink link);

    @DELETE("/shared/{sharedId}.json")
    Observable<SharedLink> deleteSharedLink(@Path("sharedId") String sharedId);
}

File: Application/src/main/kotlin/com/shareyourproxy/api/service/SharedLinkService.kt
package com.shareyourproxy.api.service

import com.shareyourproxy.api.domain.model.SharedLink
import retrofit.http.*
import rx.Observable
import java.util.*

/**
 * Add and remove [SharedLink]s.
 */
interface SharedLinkService {

    @GET("/shared.json")
    fun sharedLinks(): Observable<HashMap<String, SharedLink>>

    /**
     * add a [SharedLink]
     * @param sharedId shared link identifier
     */
    @PUT("/shared/{sharedId}.json")
    fun addSharedLink(@Path("sharedId") sharedId: String, @Body link: SharedLink): Observable<SharedLink>

    @DELETE("/shared/{sharedId}.json")
    fun deleteSharedLink(@Path("sharedId") sharedId: String): Observable<SharedLink>
}
